core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -731,21 +730,73 @@ public Result visit(Window e) {
   public Result visit(Aggregate e) {
     final Builder builder =
         visitAggregate(e, e.getGroupSet().toList(), Clause.GROUP_BY);
-    RelTrait relTrait = e.getTraitSet().getTrait(PivotRelTraitDef.instance);
-    if (relTrait != null && relTrait instanceof PivotRelTrait) {
-      if (((PivotRelTrait) relTrait).isPivotRel()) {
-        PivotRelToSqlUtil pivotRelToSqlUtil = new PivotRelToSqlUtil(POS);
-        SqlNode select =
-            pivotRelToSqlUtil.buildSqlPivotNode(e, builder, builder.select.getSelectList());
-        return result(select, ImmutableList.of(Clause.SELECT), e, null);
+    PivotRelTrait relTrait = e.getTraitSet().getTrait(PivotRelTraitDef.instance);
+    if (relTrait != null && relTrait.isPivotRel()) {
+      List<SqlNode> selectList = builder.select.getSelectList();
+      List<SqlNode> aggregateColList = new ArrayList<>();
```

Comment:
change the variable name to `aggregateInClauseFieldList`

Reply:
resolved